src/main/java/org/traccar/protocol/GranitProtocolDecoder.java

```diff
@@ -209,7 +209,7 @@ protected Object decode(Channel channel, SocketAddress remoteAddress, Object msg
                     if (buf.getUnsignedByte(buf.readerIndex()) != 0xFE) {
                         Position position = new Position(getProtocolName());
                         position.setDeviceId(deviceSession.getDeviceId());
-                        position.setTime(new Date((unixTime + i * timeIncrement) * 1000));
+                        position.setTime(new Date((unixTime + (long) i * timeIncrement) * 1000));
```

Comment:
i * increment is multiplication of two integers. Casting one of them to long prevents integer overflow (long overflow is still possible though).